client/utils/auth.py

```diff
@@ -169,53 +169,60 @@ def perform_oauth(assignment, code_fn):
     update_storage(access_token, expires_in, refresh_token)
     return access_token
 
-def authenticate(assignment, force=False):
+def server_url(cmd_args):
```

Comment:
I'm fine with this as is - but should this be a property of Settings? That way protocols could use `cmd_args.server_url` instead of importing this function (I suppose could use assignment.server_url like backup.py does) 

Keeping the settings object simple is fine too.